akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -37,9 +37,14 @@ import akka.actor.typed.Behavior.UntypedBehavior
         untyped.child(child.path.name) match {
           case Some(`c`) ⇒
             untyped.stop(c)
-            true
+          case _ if child.path.parent != self.path ⇒
```

Comment:
👍

Reply:
Oh, wait, if the ref has the same name as a child this will break